cdap-app-fabric/src/main/java/co/cask/cdap/internal/provision/ProvisioningService.java

```diff
@@ -100,36 +95,13 @@ protected void startUp() throws Exception {
     reloadProvisioners();
     // TODO: CDAP-13246 check ProvisionerDataset to find any operations that are supposed to be in progress and
     // pick up where they left off
-    this.executorService = Executors.newCachedThreadPool(
-      new ThreadFactoryBuilder().setDaemon(true).setNameFormat("provisioning-service").build());
   }
 
   @Override
   protected void shutDown() throws Exception {
-    LOG.info("Stopping {}", getClass().getSimpleName());
-    try {
-      // Shutdown the executor, which will issue an interrupt to the running thread.
-      executorService.shutdownNow();
-      executorService.awaitTermination(5, TimeUnit.SECONDS);
-    } catch (InterruptedException ie) {
-      // Ignore it.
-    } finally {
-      if (!executorService.isTerminated()) {
-        executorService.shutdownNow();
-      }
-    }
     LOG.info("Stopped {}", getClass().getSimpleName());
   }
 
-  /**
-   * Execute a provisioning runnable.
-   *
-   * @param runnable the runnable to execute
-   */
-  public void execute(ProvisioningTask runnable) {
```

Comment:
In that case, the `ProvioningTask` returned should be using the execute internally instead of requiring the caller to call this method to provide better isolation. I'll update this.

BTW, for the stopping scenario, it probably could get more complicated, as the stop can happen any time (including during provisioning or launching). We probably need to refactor that to be handled via TMS and the program notification subscriber.